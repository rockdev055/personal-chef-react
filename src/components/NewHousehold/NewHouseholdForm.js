import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button, Form, Segment, Checkbox } from 'semantic-ui-react';

import styled from 'styled-components';

export const StyledForm = styled.div`
  ul {
    max-width: 800px;
    margin: 0 auto;
  }
  li {
    display: flex;
    flex-wrap: wrap;
    alignitems: center;
    justifycontent: space-between;
    marginbottom: 20px;
  }
  label {
    flex: 1 0 120px;
    max-width: 220px;
  }
  input {
    flex: 1 0 220px;
    padding: 15px;
    border-radius: 5px;
    border: 2px solid gray;
    :focus: {
      outline: 'none';
    }
  }

  button {
    margin-left: auto;
    padding: 8px 16px;
    border: none;
    background: #333;
    color: #f2f2f2;
    text-transform: uppercase;
    letter-spacing: 0.09em;
    border-radius: 2px;
  }
`;

class NewHouseholdForm extends Component {
  state = { name: '', client: false, address: '', monthly_rate: '' };

  onChange = e => {
    console.log(e.target.type);
    const { target } = e;
    const { name } = target;
    const value = target.type === 'checkbox' ? target.checked : target.value;
    this.setState({ [name]: value });
  };

  render() {
    const { onSubmit } = this.props;
    const { name, address, monthly_rate: monthlyRate, client } = this.state;
    return (
      <Form
        data-testid="household-form"
        size="large"
        onSubmit={e => {
          e.preventDefault();
          onSubmit(this.state);
        }}
      >
        <Segment stacked>
          <Form.Input
            placeholder="Family Name"
            name="name"
            onChange={this.onChange}
            type="text"
            id="name"
            value={name}
          />
          <Form.Input
            name="address"
            placeholder="Address"
            id="address"
            type="text"
            value={address}
            onChange={this.onChange}
          />
          <Form.Input
            name="monthly_rate"
            placeholder="Potential Monthly Rate"
            id="monthly_rate"
            type="text"
            value={monthlyRate}
            onChange={this.onChange}
          />
          <Form.Field>
            <label htmlFor="client">Client?</label>
            <input
              name="client"
              id="client"
              textAlign="center"
              type="checkbox"
              checked={client}
              onChange={this.onChange}
            />
          </Form.Field>
          <Button fluid size="large">
            Create Lead
          </Button>
        </Segment>
      </Form>
    );
  }
}

NewHouseholdForm.propTypes = {
  onSubmit: PropTypes.func,
};

export default NewHouseholdForm;
