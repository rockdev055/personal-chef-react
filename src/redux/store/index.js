import {
  combineReducers,
  applyMiddleware,
  createStore
} from 'redux'
import thunk from 'redux-thunk'
import auth from '../modules/Auth/reducer'
import households from '../modules/Households/reducer'
import members from '../modules/Members/reducer'
import meals from '../modules/Meals/reducer'
import { reducer as form } from 'redux-form'

const rootReducer = combineReducers({
  auth,
  form,
  households,
  meals,
  members
})

const middleware = [thunk]

export default createStore(rootReducer, window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(), applyMiddleware(...middleware))


