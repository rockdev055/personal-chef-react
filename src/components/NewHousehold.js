import React, { Component } from 'react'
import NewHouseholdForm from './NewHouseholdForm'

class NewHousehold extends Component {

  submit = (values) => {
    console.log(values)
  }

  render() {
    return (
      <div>
        <h2>
          Create A New Household
        </h2>
        <NewHouseholdForm onSubmit={this.submit}/>
      </div>
    )
  }
}

export default NewHousehold