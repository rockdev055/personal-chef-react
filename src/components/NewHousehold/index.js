import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Grid, Header, Message } from 'semantic-ui-react';

import NewHouseholdForm from './NewHouseholdForm';

import { createHousehold } from '../../redux/modules/Households/actions';

class NewHousehold extends Component {
  submit = values => {
    const { createHousehold, history } = this.props;
    createHousehold(values, history);
  };

  render() {
    return (
      <Grid textAlign="center" style={{ height: '100vh' }} verticalAlign="middle">
        <Grid.Column style={{ maxWidth: 600 }}>
          <Header as="h2" textAlign="center" data-testid="page-header">
            Create A New Family
          </Header>
          <NewHouseholdForm onSubmit={this.submit} />
        </Grid.Column>
      </Grid>
    );
  }
}

NewHousehold.propTypes = {
  createHousehold: PropTypes.func,
  history: PropTypes.object,
};

export default connect(null, { createHousehold })(NewHousehold);
