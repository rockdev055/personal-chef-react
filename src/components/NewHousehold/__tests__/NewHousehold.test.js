import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../../redux/store';
import NewHousehold from '../index';

afterEach(cleanup);

describe('<NewHousehold /> component', () => {
  it('matches the snapshop', () => {
    const { container } = render(
      <Provider store={store}>
        <NewHousehold />
      </Provider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <NewHousehold />
      </Provider>
    );

    expect(getByTestId('page-header').textContent).toBe('Create A New Family');
  });

  it('renders the new household form', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <NewHousehold />
      </Provider>
    );

    expect(getByTestId('household-form')).toBeTruthy();
  });

  it('submits the form correctly', () => {
    const { getByText } = render(
      <Provider store={store}>
        <NewHousehold />
      </Provider>
    );

    fireEvent.click(getByText('Create Lead'));
  });
});
