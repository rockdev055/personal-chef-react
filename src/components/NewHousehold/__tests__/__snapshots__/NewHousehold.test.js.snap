// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewHousehold /> component matches the snapshop 1`] = `
<div
  data-reactroot=""
>
  <h2
    data-testid="page-header"
  >
    Create A New Family
  </h2>
  <div
    class="sc-bdVaJa kwuwfT"
  >
    <form
      data-testid="household-form"
    >
      <ul>
        <li>
          <label
            for="name"
          >
            <!-- react-text: 8 -->
            Family Name
            <!-- /react-text -->
            <input
              type="text"
              name="name"
              id="name"
              value=""
            />
          </label>
        </li>
        <li>
          <label
            for="address"
          >
            <!-- react-text: 12 -->
            Address
            <!-- /react-text -->
            <input
              type="text"
              name="address"
              id="address"
              value=""
            />
          </label>
        </li>
        <li>
          <label
            for="monthly_rate"
          >
            <!-- react-text: 16 -->
            Potential Monthly Rate
            <!-- /react-text -->
            <input
              type="text"
              name="monthly_rate"
              id="monthly_rate"
              value=""
            />
          </label>
        </li>
        <li>
          <label
            for="client"
          >
            <!-- react-text: 20 -->
            Client?
            <!-- /react-text -->
            <input
              type="checkbox"
              name="client"
              id="client"
            />
          </label>
        </li>
        <li>
          <button
            type="submit"
          >
            Create Lead
          </button>
        </li>
      </ul>
    </form>
  </div>
</div>
`;
